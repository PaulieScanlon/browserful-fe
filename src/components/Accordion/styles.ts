import styled from 'react-emotion';
import { common, colours, spaceMd } from '../../theme';

interface IProps {
  maxHeight: string;
}

export const AccordionWrapper = styled.div({
  label: 'acorrdion-wrapper',
  position: 'relative',
  width: '100%',
  boxShadow: common.materialBoxShadow1
});

export const Input = styled.input<IProps>(
  {
    label: 'input',
    position: 'absolute',
    top: '-10px',
    left: '-10px',
    display: 'none',

    ':checked + label': {
      '&:after': {
        content: `""`,
        width: '14px',
        height: '14px',
        borderWidth: '4px',
        borderColor: `${colours.pink}`,
        backgroundColor: `${colours.white}`,
        color: colours.pink
      }
    }
  },
  ({ maxHeight }) => ({
    ':checked ~ div': {
      maxHeight: maxHeight
    }
  })
);

export const Label = styled.label({
  label: 'label',
  position: 'relative',
  display: 'block',
  backgroundColor: 'white',
  cursor: 'pointer',
  padding: `${spaceMd}px`,
  color: colours.greyDark,
  borderBottom: `1px solid ${colours.greyLight}`,
  transition: `${common.transition}`,
  boxSizing: 'border-box',
  ':after': {
    content: `""`,
    position: 'absolute',
    width: '20px',
    height: '20px',
    borderWidth: '1px',
    borderStyle: 'solid',
    borderRadius: '100%',
    right: `${spaceMd}px`,
    textAlign: 'center',
    borderColor: `${colours.greyLight}`,
    transition: `${common.transition}`,
    color: colours.greyLight
  }
});

export const ContentWrapper = styled.div({
  label: 'content-wrapper',
  overflow: 'hidden',
  maxHeight: '0px',
  transition: `${common.transition}`,
  padding: `0px ${spaceMd}px`,
  backgroundColor: colours.offWhite,
  color: colours.greyDark
});

export const Content = styled.div({
  label: 'content',
  padding: `${common.paddingTB} 0px`
});
