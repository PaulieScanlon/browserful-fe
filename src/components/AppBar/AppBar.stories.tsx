import * as React from 'react';
import { storiesOf } from '@storybook/react';
import { withInfo } from '@storybook/addon-info';
import { action } from '@storybook/addon-actions';

import { spaceLg } from '../../theme';

import { AppBar } from './AppBar';
import { Button } from '../Button';

const stories = storiesOf('AppBar', module);

const onClick = event => {
  action('onClick')(event.currentTarget);
};

stories.add(
  'default usage',
  withInfo(
    'AppBar houses the Browserful Logo and other things... The disableLink prop disables the a `Link` which is part of `next-routes'
  )(() => <AppBar disableLink={true} />)
);

stories.add(
  'showLogo',
  withInfo('The showLogo prop controls the visibility of the Browserful logo')(
    () => <AppBar showLogo={false} />
  )
);

stories.add(
  'children: HTML',
  withInfo('Children can be HTML')(() => (
    <AppBar>
      <span style={{ marginRight: `${spaceLg}px` }}>Hi, I'm an HTML child</span>
    </AppBar>
  ))
);

stories.add(
  'children: React',
  withInfo('Children can be React components')(() => (
    <AppBar>
      <span style={{ marginRight: `${spaceLg}px` }}>
        <Button onClick={event => onClick(event)}>Click me</Button>
      </span>
    </AppBar>
  ))
);
