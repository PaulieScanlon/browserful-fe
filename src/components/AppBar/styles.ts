import styled from 'react-emotion';

import { colours, common } from '../../theme';

export const AppBarWrapper = styled.header({
  label: 'app-bar-wrapper',
  display: 'flex',
  alignItems: 'center',
  justifyContent: 'space-between',
  backgroundColor: colours.white,
  boxShadow: common.materialBoxShadow1,
  minHeight: common.appBarHeight
});
