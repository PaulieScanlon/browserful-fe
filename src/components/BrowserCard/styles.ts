import styled from 'react-emotion';
import { common, colours, spaceMd } from '../../theme';

interface IProps {
  isExpanded?: boolean;
}

export const BrowserCardWrapper = styled.div({
  label: 'browser-card-wrapper',
  backgroundColor: colours.white,
  boxShadow: common.materialBoxShadow1
});

export const BrowserCardHeader = styled.div<IProps>(
  {
    label: 'browser-card-header',
    display: 'flex',
    justifyContent: 'space-between',
    alignItems: 'center',
    margin: `0 ${spaceMd}px`,
    padding: `${spaceMd}px 0px`,
    borderBottomWidth: '1px',
    borderBottomStyle: 'solid'
  },
  ({ isExpanded }) => ({
    borderBottomColor: isExpanded ? colours.greyLight : colours.white
  })
);

export const BrowserCardHeaderSection = styled.div({
  label: 'browser-card-header-section',
  display: 'inline-flex',
  alignItems: 'center'
});

export const BrowserCardBody = styled.div<IProps>(
  {
    label: 'browser-card-body',
    overflow: 'hidden'
  },
  ({ isExpanded }) => ({
    height: isExpanded ? '100%' : '0px',
    padding: isExpanded ? `${spaceMd}px ${spaceMd}px` : `0px ${spaceMd}px`
  })
);
