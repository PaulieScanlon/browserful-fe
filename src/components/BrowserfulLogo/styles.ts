import styled from 'react-emotion';

import { common, spaceLg } from '../../theme';

export const LogoWrapper = styled.div({
  label: 'logo-wrapper',
  display: 'inline-flex',
  alignItems: 'center',
  marginLeft: `${spaceLg}px`
});

export const SvgWrapper = styled.span({
  label: 'svg-wrapper',
  display: 'inline-flex',
  marginRight: '5px'
});
