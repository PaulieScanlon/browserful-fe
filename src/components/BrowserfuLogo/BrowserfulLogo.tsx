import * as React from 'react';

import { colours } from '../../theme';

import { LogoWrapper } from './styles';

interface IProps {
  theme?: 'pink' | 'teal';
}

export const BrowserfulLogo: React.SFC<IProps> = ({ theme }: IProps) => {
  return (
    <LogoWrapper>
      <div>BrowserfulLogo {theme}</div>

      <svg
        style={{ fill: `${colours[theme]}` }}
        x="0px"
        y="0px"
        width="23.5"
        height="23.5"
      >
        <circle cx="9.6" cy="9.6" r="9.6" />
        <g>
          <circle cx="20.3" cy="20.3" r="3.2" />
        </g>
      </svg>
    </LogoWrapper>
  );
};

BrowserfulLogo.defaultProps = { theme: 'pink' };
