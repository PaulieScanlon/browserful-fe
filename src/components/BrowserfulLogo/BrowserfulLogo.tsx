import * as React from 'react';

import { colours } from '../../theme';

import { H5 } from '../../typography';
import { LogoWrapper, SvgWrapper } from './styles';

interface IProps {
  showText?: boolean;
  fontColour?: string;
  logoColour?: string;
}

export const BrowserfulLogo: React.SFC<IProps> = ({
  showText,
  fontColour,
  logoColour
}: IProps) => {
  return (
    <LogoWrapper>
      <SvgWrapper>
        <svg
          style={{ fill: `${logoColour}` }}
          x="0px"
          y="0px"
          width="23.5"
          height="23.5"
        >
          <circle cx="9.6" cy="9.6" r="9.6" />
          <g>
            <circle cx="20.3" cy="20.3" r="3.2" />
          </g>
        </svg>
      </SvgWrapper>

      {showText && (
        <H5 display="inline-flex" fontColour={fontColour}>
          Browserful
        </H5>
      )}
    </LogoWrapper>
  );
};

BrowserfulLogo.defaultProps = {
  showText: true,
  logoColour: `${colours.pink}`,
  fontColour: `${colours.greyMid}`
};
