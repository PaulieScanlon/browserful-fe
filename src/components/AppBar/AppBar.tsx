import * as React from 'react';

import { BrowserfulLogo } from '../BrowserfulLogo';
import { AppBarWrapper } from './styles';
import Link from 'next/link';

interface IProps extends React.Props<ChildNode> {
  showLogo?: boolean;
  fixed?: boolean;
  disableLink?: boolean;
}

export const AppBar: React.SFC<IProps> = ({
  showLogo,
  fixed,
  children
}: IProps) => {
  return (
    <AppBarWrapper fixed={fixed}>
      {showLogo && (
        <Link href="/">
          <a>
            <BrowserfulLogo />
          </a>
        </Link>
      )}
      {children}
    </AppBarWrapper>
  );
};

AppBar.defaultProps = {
  showLogo: true,
  fixed: false,
  disableLink: false
};
