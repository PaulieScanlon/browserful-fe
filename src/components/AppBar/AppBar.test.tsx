import * as React from 'react';
import { shallow, mount } from 'enzyme';

import { AppBar } from './AppBar';
import { AppBarWrapper } from './styles';
import { BrowserfulLogo } from '../BrowserfulLogo';

const shallowDefault = props => {
  return shallow(<AppBar {...props} />);
};

describe('<AppBar />', () => {
  it('renders correctly', () => {
    const wrapper = shallowDefault({});
    expect(wrapper).toHaveLength(1);
    expect(wrapper).toMatchSnapshot();
  });
  it('has a <AppBarWrapper>', () => {
    const wrapper = shallowDefault({});
    const appBarWrapper = wrapper.find(AppBarWrapper);
    expect(appBarWrapper).toHaveLength(1);
  });
  it('hides logo if showLogo:false', () => {
    const wrapper = shallowDefault({ showLogo: false });
    const browserfulLogo = wrapper.find(BrowserfulLogo);
    expect(browserfulLogo).toHaveLength(0);
  });
  it('renders HTML children', () => {
    const wrapper = shallow(
      <AppBar>
        <span>child</span>
      </AppBar>
    );
    expect(wrapper.children.length).toEqual(1);
  });
  it.skip('renders React child', () => {
    const wrapper = shallow(<AppBar />);
    expect(wrapper.children.length).toEqual(1);
  });
});
