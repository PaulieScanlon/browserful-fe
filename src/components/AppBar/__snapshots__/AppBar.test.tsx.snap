// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppBar /> renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AppBar
    disableLink={false}
    fixed={false}
    showLogo={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Link
          href="/"
        >
          <a>
            <BrowserfulLogo
              fontColour="#404040"
              logoColour="#ff5ca3"
              showText={true}
            />
          </a>
        </Link>,
        undefined,
      ],
      "fixed": false,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <a>
            <BrowserfulLogo
              fontColour="#404040"
              logoColour="#ff5ca3"
              showText={true}
            />
          </a>,
          "href": "/",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <BrowserfulLogo
              fontColour="#404040"
              logoColour="#ff5ca3"
              showText={true}
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "fontColour": "#404040",
              "logoColour": "#ff5ca3",
              "showText": true,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "a",
        },
        "type": [Function],
      },
      undefined,
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Link
            href="/"
          >
            <a>
              <BrowserfulLogo
                fontColour="#404040"
                logoColour="#ff5ca3"
                showText={true}
              />
            </a>
          </Link>,
          undefined,
        ],
        "fixed": false,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <a>
              <BrowserfulLogo
                fontColour="#404040"
                logoColour="#ff5ca3"
                showText={true}
              />
            </a>,
            "href": "/",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <BrowserfulLogo
                fontColour="#404040"
                logoColour="#ff5ca3"
                showText={true}
              />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "fontColour": "#404040",
                "logoColour": "#ff5ca3",
                "showText": true,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "a",
          },
          "type": [Function],
        },
        undefined,
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
